test/system/flows/edit_solution_test.rb

```diff
@@ -0,0 +1,57 @@
+require "application_system_test_case"
+require_relative "../../support/capybara_helpers"
+
+module Components
+  module Student
+    class EditorTest < ApplicationSystemTestCase
+      include CapybaraHelpers
+
+      test "user runs tests" do
+        user = create :user
+        create :user_auth_token, user: user
+        solution = create :concept_solution, user: user
+
+        visit edit_solution_path(solution.uuid)
+        click_on "Run tests"
+        wait_for_submission
+        2.times { wait_for_websockets }
+        test_run = create :submission_test_run,
```

Comment:
Is this not handled by the controller?

Reply:
No, the test run is created once it comes back from the test-running infrastructure.

Reply:
Cool. I think this should be documented, because it's not clear that one would require a side-effect in the test to make the test pass.